File: c/cz/mg/vulkan/PFN_vkCreatePipelineLayout.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCreatePipelineLayout_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pPipelineLayout, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkCreatePipelineLayout f = (PFN_vkCreatePipelineLayout)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkPipelineLayoutCreateInfo*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkPipelineLayout*)jniLongToPointer(pPipelineLayout))
    );
}

File: cz/mg/vulkan/PFNvkCreatePipelineLayout.java
package cz.mg.vulkan;

public class PFNvkCreatePipelineLayout extends VkFunctionPointer {
    public PFNvkCreatePipelineLayout() {
    }

    public PFNvkCreatePipelineLayout(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCreatePipelineLayout(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkCreatePipelineLayout(long value) {
        setValue(value);
    }

    public PFNvkCreatePipelineLayout(VkInstance instance) {
        super(instance, new VkString("vkCreatePipelineLayout"));
    }

    public void call(VkDevice device, VkPipelineLayoutCreateInfo pCreateInfo, VkAllocationCallbacks pAllocator, VkPipelineLayout pPipelineLayout, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), pCreateInfo != null ? pCreateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pPipelineLayout != null ? pPipelineLayout.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getNullAddress());
    }

    private static native void call(long vkaddress, long device, long pCreateInfo, long pAllocator, long pPipelineLayout, long rval);
}
